### PR TITLE
Fix/concatenated

### LINKED ISSUE #395: Error in sequence_file_in
```cpp
struct my_traits : public sequence_file_in_default_traits_dna
{
    using sequence_alphabet = dna4;                        // instead of dna5

    template <typename alph>
    using sequence_container = bitcompressed_vector<alph>; // must be defined as a template!
};

sequence_file_in<my_traits> fin{"/tmp/my.fasta"};

//...
```

Produces errors about sequence_container_concept, among other strange things.

### PR DESCRIPTION
**Pull request by h-2**
fixes #395 

(actually a bunch of different problems)

### CODE REVIEW

**Comment by h-2** [comment]
I advise going through the commits one-by-one (they are very small individually)

File: include/seqan3/range/container/concatenated_sequences.hpp [hunk 1]
```diff
@@ -923,9 +926,18 @@ class concatenated_sequences
             value_len = std::distance(ranges::begin(value), ranges::end(value));
 
         data_values.reserve(data_values.size() + count * value_len);
+        auto placeholder = ranges::view::repeat_n(value_type_t<rng_type>{}, count * value_len)
+                         | view::common;
```
Comment (marehr) [inline include/seqan3/range/container/concatenated_sequences.hpp]: The commit message is `[FIX] ensure that ranges passed on are actually ForwardRanges `, but I can't find where this is actually required in your first commit.

Comment (rrahn) [inline include/seqan3/range/container/concatenated_sequences.hpp]: The pipe symbol on the previous line

File: include/seqan3/alphabet/concept.hpp [hunk 2]
```diff
@@ -91,7 +91,7 @@ namespace seqan3
 template <typename t>
 concept semi_alphabet_concept = std::Regular<std::remove_reference_t<t>> &&
                                      standard_layout_concept<std::remove_reference_t<t>> &&
-                                     trivial_concept<std::remove_reference_t<t>> &&
+//                                      trivial_concept<std::remove_reference_t<t>> &&
```
Comment (rrahn) [inline include/seqan3/alphabet/concept.hpp]: why is this necessary?

File: include/seqan3/range/container/concatenated_sequences.hpp (hunk 1 above)
Comment (h-2) [reply include/seqan3/range/container/concatenated_sequences.hpp]: this calls the same functions on the underlying range and they require forward

File: include/seqan3/alphabet/concept.hpp (hunk 2 above)
Comment (h-2) [reply include/seqan3/alphabet/concept.hpp]: see #444

**Pull request merged**